internal/services/oracledatabase/exadata_infrastructure_resource_test.go

```diff
@@ -0,0 +1,214 @@
+	})
+}
+
+func TestExaInfra_allFields(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.allFields(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestExaInfra_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestExaInfra_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (a ExadataInfraResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+
+%s
+
+resource "azurerm_oracledatabase_exadata_infrastructure" "test" {
+  name                = "OFakeacctest%[2]d"
+  location            = "%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+  compute_count       = "2"
+  display_name        = "OFakeacctest%[2]d"
+  shape               = "Exadata.X9M"
+  storage_count       = "3"
+  zones               = ["3"]
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a ExadataInfraResource) allFields(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+
+%s
+
+resource "azurerm_oracledatabase_exadata_infrastructure" "test" {
+  name                = "OFakeacctest%[2]d"
+  location            = "%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+  compute_count       = "2"
+  display_name        = "OFakeacctest%[2]d"
+  shape               = "Exadata.X9M"
+  storage_count       = "3"
+  zones               = ["3"]
+  customer_contacts   = ["test@test.com"]
+  maintenance_window {
+    custom_action_timeout_in_mins    = 0
+    days_of_week                     = ["Monday"]
+    hours_of_day                     = [1]
+    is_custom_action_timeout_enabled = false
+    is_monthly_patching_enabled      = true
+    months                           = ["January"]
+    weeks_of_month                   = [2]
+    lead_time_in_weeks               = 1
+    patching_mode                    = "Rolling"
+    preference                       = "NoPreference"
+  }
+  tags = {
```

Comment:
we should make this tidier
```suggestion
  customer_contacts   = ["test@test.com"]

  maintenance_window {
    custom_action_timeout_in_mins    = 0
    days_of_week                     = ["Monday"]
    hours_of_day                     = [1]
    is_custom_action_timeout_enabled = false
    is_monthly_patching_enabled      = true
    months                           = ["January"]
    weeks_of_month                   = [2]
    lead_time_in_weeks               = 1
    patching_mode                    = "Rolling"
    preference                       = "NoPreference"
  }

  tags = {
```